cache-filter/src/utils.rs

```diff
@@ -106,6 +106,18 @@ pub fn do_auth_call<C: proxy_wasm::traits::HttpContext>(
             return in_request_failure(ctx, filter);
         }
     };
+    // Pause the current request to wait for the response from 3scale
+    Action::Pause
+}
 
-    Action::Continue
+pub fn period_from_response(res_period: &ResponsePeriod) -> Period {
+    match res_period {
+        ResponsePeriod::Minute => Period::Minute,
+        ResponsePeriod::Hour => Period::Hour,
+        ResponsePeriod::Day => Period::Day,
+        ResponsePeriod::Week => Period::Week,
+        ResponsePeriod::Month => Period::Month,
+        ResponsePeriod::Year => Period::Year,
+        ResponsePeriod::Eternity => Period::Eternity,
+    }
 }
```

Comment:
Have you tried to `impl<'a> From<&'a ResponsePeriod> for Period { ... }`? The you can simply have stuff like:

```rust
let response_period: ResponsePeriod = [...];
// get a Period out of the ResponsePeriod (without consuming it, ie. by reference)
let period = Period::from(&response_period);
```

I'd also probably place this conversion next to the place where `Period` is defined.